connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -829,6 +878,34 @@ private boolean startConnector(String connectorName) {
         return started;
     }
 
+    private Callable<Void> getConnectorCallable(final String connectorName, final ConnectorStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startConnector(connectorName);
+                        } catch (ConfigException e) {
```

Comment:
Does this need the `catch (Throwable t)` treatment as well? It could fail in ways other than a `ConfigException`, right?